src/controllers/recipe.js

```diff
@@ -1,76 +1,112 @@
 import express from 'express';
 import request from 'request';
-import yummly from '../../config/yummly.js';
+import _ from 'lodash';
+import yummlyConfig from '../../config/yummly.js';
 
 const router = express();
-const YUMMLY_BASE_URL = `http://api.yummly.com/v1/api/`;
-const YUMMLY_KEYS = `_app_id=${yummly.id}&_app_key=${yummly.key}`;
-const YUMMLY_PARAMS = `maxResult=8`;
-const YUMMLY_RECIPE_SEARCH_URL = `${YUMMLY_BASE_URL}recipes?${YUMMLY_KEYS}&${YUMMLY_PARAMS}`;
 
 export function routes() {
     return router;
 };
 
-router.get('/recipes/featured', (req, res) => {
-    request({uri: YUMMLY_RECIPE_SEARCH_URL,
-        timeout: 15000,
-        followRedirect: true,
-        maxRedirects: 10,
-        json: true
-    }, (error, response, body) => {
-        if (!error && response.statusCode === 200) {
-            res.status(200);
-            res.send({status: 'success', data: body});
-        } else {
-            console.log(`ER: ${error}, ${response.statusCode}`);
-            res.status(400);
-            res.send({
-                status: 'failure',
-                error: `${response.statusCode}: Unable to retrieve recipes.`
-            });
+export function yummly(params) {
+    const {path, queryParams = {}, jsonp = false, body = null} = params;
+
+    return new Promise((resolve, reject) => {
+        let requestParams = {
+            baseUrl: yummlyConfig.baseUrl,
+            uri: path,
+            qs: _.assign({}, queryParams, yummlyConfig.queryParams),
+            timeout: 5000,
+            followRedirects: true,
+            maxRedirects: 10,
+            json: !jsonp
+        };
+
+        if (body) {
+            requestParams = _.assign(requestParams, {body});
         }
+
+        request(requestParams, (err, response, resBody) => {
+            if (err) {
+                console.error(err);
+                return reject(err);
+            } else if (response.statusCode >= 400) {
+                console.error(response);
+                return reject(response);
+            }
+
+            if (jsonp) {
+                const jsonStart = resBody.indexOf('({');
+                const jsonEnd = resBody.indexOf('})');
+                const json = JSON.parse(resBody.substring(jsonStart + 1, jsonEnd + 1));
+
+                resolve(json);
+            } else {
+                resolve(resBody);
+            }
+        });
     });
-});
+};
 
-router.get('/recipes/search/:searchTerm', (req, res) => {
-    request({uri: `${YUMMLY_RECIPE_SEARCH_URL}&q=${req.params.searchTerm}`,
-        timeout: 15000,
-        followRedirect: true,
```

Comment:
Yummly says that it's possible for there to be redirects, and we should be able to handle them properly.